server/services/v1/api.go

```diff
@@ -445,3 +445,62 @@ func (s *apiService) Events(r *api.EventsRequest, stream api.Tigris_EventsServer
 
 	return nil
 }
+
+func (s *apiService) Publish(ctx context.Context, r *api.PublishRequest) (*api.PublishResponse, error) {
+	var wo *api.WriteOptions = nil
+	if r.GetOptions() != nil {
+		wo = r.GetOptions().WriteOptions
+	}
+
+	resp, err := s.Insert(ctx, &api.InsertRequest{
```

Comment:
Instead of Insert, we need to use Replace. Insert also does check whether the row already exists or not, if we need to generate the id on the server side then we can guarantee that the row doesn't exist. Also, the default id is UUID which is what we want to use in this case? Also, did you think about the use-case of supporting timestamp-based queries from this collection?